### DISCUSSION TITLE
What should the window size of the game be?

### DISCUSSION THREAD

**Discussion by Anonymous4045** [letsbuilda/tower#23] [Q&A]
Currently, each block is 32x32 pixels, and the width is 30 blocks wide by 20 blocks tall, making the window 960x640. However, we recently decided to make each block 64x64 to give more detail to the sprites. So do we then just keep the screen the same pixel size but have less blocks (15x10) or do we increase the window size?

We should keep in mind that we want the game to be playable for the most people possible so we can get more ratings, so being compatible with smaller screens can be important.

**Comment by Tebarem** [comment]
Would be nice to keep it dynamic to any windowsize. Scale everything up and down. the block spirtes should defo be 64x64 tho imo

**Reply by Anonymous4045** [reply]
So have all the sprites change size depending on the screen size? That could work, I'm pretty sure Arcade has a way of doing that. Question then would be what size should be considered the "ideal" size, where the scale is 1x, and how many blocks wide/tall?

**Reply by Tebarem** [reply]
well if we have a 30x30 tilemap we just do 64x30 1920x1920 would be the ideal starting rez (might be overkill)

> So have all the sprites change size depending on the screen size? That could work, I'm pretty sure Arcade has a way of doing that. Question then would be what size should be considered the "ideal" size, where the scale is 1x, and how many blocks wide/tall?

**Reply by Tebarem** [reply]
might wanna change the height of the tilemap to something resonable

**Reply by Anonymous4045** [reply]
> might wanna change the height of the tilemap to something resonable

The height at the moment is 20 blocks. If we change it to 16 and set the width at 20, we can have the default size be 1280x1024px, which will fit on most people's screens. So I think having the new tilemap be 16 blocks high by 20 blocks wide would be best for most people, and those with larger and smaller screens can still view it in fullscreen with scaling.

**Reply by Tebarem** [reply]
> > might wanna change the height of the tilemap to something resonable
> 
> The height at the moment is 20 blocks. If we change it to 16 and set the width at 20, we can have the default size be 1280x1024px, which will fit on most people's screens. So I think having the new tilemap be 16 blocks high by 20 blocks wide would be best for most people, and those with larger and smaller screens can still view it in fullscreen with scaling.

This sounds good but i think the width is too small. We can do attempt a different 16:10 rez or a 16:9 rez

**Reply by Anonymous4045** [reply]
> This sounds good but i think the width is too small. We can do attempt a different 16:10 rez or a 16:9 rez

What about 24x15 blocks then? That's a 16:10 ratio, and it'd be 1536x960px, which is a slightly weird res but it could work.
